src/app/games/hexodia/tests/HexodiaRules.spec.ts

```diff
@@ -0,0 +1,441 @@
+                [_, _, _, _, _, _, _, _, _, _, _, N, N],
+                [_, _, _, _, _, _, _, _, _, _, N, N, N],
+                [_, _, _, _, _, _, _, _, _, N, N, N, N],
+                [_, _, _, _, _, _, _, _, N, N, N, N, N],
+                [_, _, _, _, _, _, _, N, N, N, N, N, N],
+            ], 1);
+            RulesUtils.expectMoveSuccess(rules, state, move, expectedState, defaultConfig);
+        });
+
+        it('should refuse move that drops two pieces on first turn', () => {
+            // Given the first turn
+            const state: HexodiaState = HexodiaRules.get().getInitialState(defaultConfig);
+            // When dropping two pieces
+            const move: HexodiaMove = HexodiaMove.of([new Coord(11, 11), new Coord(10, 10)]);
+            // Then the attempt would have throw
+            function tryDoubleDropOnFirstTurn(): void {
+                rules.isLegal(move, state, defaultConfig);
+            }
+            TestUtils.expectToThrowAndLog(tryDoubleDropOnFirstTurn, 'HexodiaMove should only drop one piece at first turn');
+        });
+
+    });
+
+    describe('next turns', () => {
+
+        it('should forbid move where second coord is out of range', () => {
+            // Given any board on second turn
+            const state: HexodiaState = new HexodiaState([
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, O, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+                [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            ], 1);
+
+            // When doing a move who'se second coord is out of range
+            const move: HexodiaMove = HexodiaMove.of([new Coord(0, 0), new Coord(-1, -1)]);
+
+            // Then it should fail
```

Comment:
Then the move should be illegal

Reply:
// Then the move should be illegal x 283
RulesUtils.expectMoveFailure x 285 !
FOR L'UNIFORME A PRESIN (j'ai du changer que Hexodia)